android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -333,18 +352,20 @@ public void onRefresh() {
   }
 
   private void emptyGroupIdQueue() {
-    List<PwPair> pwPairs = new ArrayList<>();
+    if (!SwipeDismissListViewTouchListener.isLocked()) {
+      List<PwPair> pwPairs = new ArrayList<>();
```

Comment:
hm, you can't just not empty the queue if it's locked...you'll have to empty it later.  Maybe you can empty it and just not call notifyDatasetchanged (and then call notifyDataSetChanged in onDismiss).  The same applies above for adding new urls.